x/perp/v2/integration/assertion/event.go

```diff
@@ -14,18 +17,185 @@ import (
 	types "github.com/NibiruChain/nibiru/x/perp/v2/types"
 )
 
+// --------------------------------------------------
+// Action Functions
+// --------------------------------------------------
+
+var _ action.Action = (*containsLiquidateEvent)(nil)
+var _ action.Action = (*positionChangedEventShouldBeEqual)(nil)
+
+// PositionChangedEventShouldBeEqual checks that the position changed event is
+// equal to the expected event.
+func PositionChangedEventShouldBeEqual(
+	expectedEvent *types.PositionChangedEvent,
+) action.Action {
+	return positionChangedEventShouldBeEqual{
+		ExpectedEvent: expectedEvent,
+	}
+}
+
+// ContainsLiquidateEvent checks if a typed event (proto.Message) is contained in the
+// event manager of the app context.
+func ContainsLiquidateEvent(
+	expectedEvent types.LiquidationFailedEvent,
+) action.Action {
+	return containsLiquidateEvent{
+		ExpectedEvent: expectedEvent,
+	}
+}
+
+// ProtoToJson converts a proto message into a JSON string using the proto codec.
+// A codec defines a functionality for serializing other objects. The proto
+// codec provides full Protobuf serialization compatibility.
+func ProtoToJson(protoMsg proto.Message) (jsonOut string, err error) {
+	protoCodec := codec.NewProtoCodec(codectypes.NewInterfaceRegistry())
+	var jsonBz json.RawMessage
+	jsonBz, err = protoCodec.MarshalJSON(protoMsg)
+	return string(jsonBz), err
+}
+
+// EventEquals exports functions for comparing sdk.Events to concrete typed
+// events implemented as proto.Message instances in Nibiru.
+var EventEquals = eventEquals{}
+
+type eventEquals struct{}
+
+// EventHasAttribueValue parses the given ABCI event at a key to see if it
+// matches (contains) the wanted value.
+//
+// Args:
+//   - abciEvent: The event under test
+//   - key: The key for which we'll check the value
+//   - want: The desired value
+func EventHasAttribueValue(abciEvent sdk.Event, key string, want string) error {
```

Comment:
Does `EventHasAttribueValue` need to be exported?

Reply:
I'd honestly put it in testutil, but I wasn't sure where to put it